openwisp_utils/admin_theme/static/admin/js/menu.js

```diff
@@ -1,21 +1,118 @@
 'use strict';
+const owContainer = document.getElementById('container');
+const owMenu = document.getElementById('menu');
+const owMainContent = document.getElementById('main-content');
+const owMenuToggle = document.querySelector('.menu-toggle');
+const owHamburger = document.querySelector('.hamburger');
+
 (function () {
-  window.addEventListener('load', function(){
-    var nav = document.getElementById('main-menu'),
-        container = document.getElementById('container'),
-        header = document.getElementById('header'),
-    setHeightToMenu = function() {
-      var windowHeight = window.innerHeight,
-          pageHeight = document.body.scrollHeight || container.offsetHeight,
-          height = windowHeight;
-      if (pageHeight > windowHeight) {
-        height = pageHeight;
+  setMenu();
+  initMenuGroupClickListener();
+  initToggleMenuHandlers();
+  initAccountViewHandler();
+  initToolTipHandlers();
+})();
+
+function initMenuGroupClickListener() {
+  let mgHeads = document.querySelectorAll('.mg-head');
+  mgHeads.forEach(function (mgHead) {
+    // Handle click on menu group
+    mgHead.addEventListener('click', function (e) {
+      e.stopPropagation();
+      var currentActiveGroup = document.querySelector('.menu-group.active');
+      if (currentActiveGroup && currentActiveGroup !== mgHead.parentElement) {
+        currentActiveGroup.classList.remove('active');
       }
-      if (!nav) { return; }
-      nav.style.height = (height - header.offsetHeight) + 'px';
-      nav.style.marginTop = header.offsetHeight + 'px';
-    };
-    setHeightToMenu();
-    window.addEventListener('resize', setHeightToMenu);
+      e.target.parentElement.classList.toggle('active');
+    });
   });
-})();
+  // Handle click out side the current active menu group
+  document.addEventListener('click', function (e) {
+    var currentActiveGroup = document.querySelector('.menu-group.active');
+    if (currentActiveGroup && !currentActiveGroup.contains(e.target)) {
+      currentActiveGroup.classList.remove('active');
+    }
+  });
+}
+
+function setMenu() {
+  let openMenu = localStorage.getItem('ow-menu');
+  if (window.innerWidth > 1024) {
+    if (openMenu === null) {
+      // User visits first time. Keep open menu
+      localStorage.setItem('ow-menu', true);
+      owContainer.classList.toggle('toggle-menu');
+    } else if (openMenu === 'true') {
+      // Close the menu
+      owContainer.classList.toggle('toggle-menu');
+    }
+  }
+  setTimeout(function () {
+    // Transition fix: Add transition to menu and main content
+    // after some time.
+    if (owMenu) {
+      owMenu.style.transitionDuration = '0.3s';
```

Comment:
> 0.3s

I would prefer 0.2s or 0.1s to make it feel snappier.  Also, since there may be different preferences, you could define this `transition-duration` as a variable.